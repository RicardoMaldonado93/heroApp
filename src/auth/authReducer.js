import { types } from "../types/types";

export const authReducer = (state = {}, action = null) => {
  if (!action) return state;

  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
      };

    case types.logout:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
