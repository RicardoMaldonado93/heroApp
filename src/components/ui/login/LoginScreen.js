import React, { useContext } from 'react'
import { AuthContext } from '../../../auth/AuthContext'
import { types } from '../../../types/types';

export const LoginScreen = ({ history }) => {

    const { dispatch } = useContext( AuthContext );
    
    const handleLogin = ()=>{

        
        const action = {
            type: types.login,
            payload: {
                name : "Ricardo"
            }
            
        }
        dispatch(action);
        
        
        history.replace('/');
    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr />

            <button
                className  = "btn btn-outline-primary"
                onClick    = { handleLogin }
            >
                Sign In
            </button>

        </div>
    )
}
