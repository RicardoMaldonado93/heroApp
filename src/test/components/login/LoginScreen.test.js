import React from 'react';
import { mount, shallow } from 'enzyme';
import { LoginScreen } from '../../../components/ui/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';

describe('test with Login Screen', () => {

    const historyMock = {
        replace: jest.fn()
    }

    const contextMock =   {
        dispatch: jest.fn()
    }
    
    const wrapper = mount( 
        <AuthContext.Provider value   = { contextMock }>
            <LoginScreen history = { historyMock }/>
        </AuthContext.Provider>
    );

    test('should be match with snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });
   
    test("should be call dispatch and navigate", ()=>{
        const handleClick = wrapper.find("button").prop("onClick");

        handleClick();

        expect( contextMock.dispatch ).toHaveBeenLastCalledWith({
            type: types.login,
            payload:{
                name: "Ricardo"
            }
        });

        expect( historyMock.replace ).toHaveBeenCalled();

        localStorage.setItem("lastPath", "/dc");

        handleClick();
        expect( historyMock.replace ).toHaveBeenLastCalledWith("/dc");
    });
});
