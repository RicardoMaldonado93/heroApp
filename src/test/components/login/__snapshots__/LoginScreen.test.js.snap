// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with Login Screen should be match with snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Sign In
  </button>
</div>
`;
